### DISCUSSION TITLE
For add_routers batch endpoint - does it implement in the sync or async?

### DISCUSSION THREAD

**Discussion by SoulEvill** [langchain-ai/langserve#624] [Q&A] [answered]
I have an application that requires async batch process and currently I am serving this llm through langserve where I provide the batch endpoint to the user. It is not clear to me for this endpoint, does it implemented in async or sync manner?  

langchain chain does support abatch out of box but langserve endpoint doesnt seem to have this abatch. 

If batch endpoint is not async - any suggestions how to serve an endpoint that acommplicate async batch? 

Thanks a lot!

**Comment by eyurtsev** [comment, marked as answer]
Server code is always **async** in langserve. It'll use the async implementation of all code by default. If async implementations are not available it will fall back on sync implementations.

**Comment by SoulEvill** [comment]
Great thanks for clarity
